Formula/folly.rb

```diff
@@ -21,14 +21,29 @@ class Folly < Formula
   depends_on "glog"
   depends_on "libevent"
   depends_on "lz4"
-  # https://github.com/facebook/folly/issues/1545
-  depends_on macos: :catalina
   depends_on "openssl@1.1"
   depends_on "snappy"
   depends_on "xz"
   depends_on "zstd"
 
+  on_macos do
+    depends_on "llvm" if DevelopmentTools.clang_build_version <= 1100
+  end
+
+  fails_with :clang do
+    build 1100
+    # https://github.com/facebook/folly/issues/1545
+    cause <<-EOS
+      Undefined symbols for architecture x86_64:
+        "std::__1::__fs::filesystem::path::lexically_normal() const"
+    EOS
+  end
+
   def install
+    on_macos do
+      ENV.llvm_clang if DevelopmentTools.clang_build_version <= 1100
```

Comment:
Also tried building with `gcc`, since that's the fallback for `fails_with :clang`, but that won't work since `boost` is built using `clang`.